kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/test/TestCaseExecutor.kt

```diff
@@ -55,6 +55,11 @@ internal class TestCaseExecutor(
 
       val timeMark = MonotonicTimeSourceCompat.markNow()
 
+      // JS platforms require extra care when runTest is used, so skip it for now.
+      // Issue: https://github.com/kotest/kotest/issues/4077
+      val useCoroutineTestScope = (platform == Platform.JVM || platform == Platform.Native)
```

Comment:
Another option was `platform != Platform.JS && platform != Platform.WasmJs`, but it seems safer to restrict it to the known platforms, since the platform list could grow.

Reply:
As the `runTest` limitation is specific to `Promise`-based JS platforms, wouldn't it be safer to use `platform != Platform.JS && platform != Platform.WasmJs`? Otherwise, we could accidentally introduce the limit to future non-JS targets like `wasmWasi`.

Reply:
I've replaced it with a `when` clause to prevent both accidental restriction and accidental enabling for new platforms. Do you think it looks okay now?